docs/upgrade-to-v5.md

```diff
@@ -0,0 +1,37 @@
+# Changes in v5
+
+## About
+
+The following is a detailed collection of the changes in the major v5 release of the bson package
+for nodejs and web platforms.
+
+<!--
+1. a brief statement of what is breaking (brief as in "x will now return y instead of z", or "x is no longer supported, use y instead", etc
+2. a brief statement of why we are breaking it (bug, not useful, inconsistent behavior, better alternative, etc)
+3. if applicable, an example of suggested syntax change (can be included in (1) )
+-->
+
+### Remove reliance on Node.js Buffer
+
+> **TL;DR**: Web environments return Uint8Array; Node.js environments return Buffer
+
+For those that use the BSON library on Node.js, there is no change the BSON APIs will still return and accept instances of Node.js Buffer. Since we no longer depend on the Buffer web shim for compatibility with browsers, in non-Node.js environments a Uint8Array will be returned instead.
```

Comment:
can we mention which APIs this actually affects? and can we say in the summary instead of what? (i.e. `<what API is affected> in web environments return Uint8Array instead of <...>`

Reply:
https://jira.mongodb.org/browse/NODE-4794

That list would be looking at all the public APIs that changed in this PR and listing them out, in the interest of moving this along we will be cleaning up this guide when we reach the end of the EPIC and the diff of this PR will still be available to us then for reference. So let's defer to NODE-4794, I've added a note about creating a list of APIs changed.

Reply:
this isn't about making sure we have a well documented list, it's about knowing what these *specific* code changes actually affect, so I don't think it's out of scope of this PR to identify what will be changing. This work will still need to be done, and it is easier to do it piecemeal with the corresponding code changes than in one big block, where we might miss something that was affected due to the sheer volume of changes.